errors.go

```diff
@@ -7,46 +7,48 @@ import (
 	"github.com/Azure/go-amqp/internal/encoding"
 )
 
+// ErrCond is an AMQP defined error condition.
+// See http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-amqp-error for info on their meaning.
+type ErrCond = encoding.ErrCond
+
 // Error Conditions
 const (
 	// AMQP Errors
-	ErrorInternalError         ErrorCondition = "amqp:internal-error"
-	ErrorNotFound              ErrorCondition = "amqp:not-found"
-	ErrorUnauthorizedAccess    ErrorCondition = "amqp:unauthorized-access"
-	ErrorDecodeError           ErrorCondition = "amqp:decode-error"
-	ErrorResourceLimitExceeded ErrorCondition = "amqp:resource-limit-exceeded"
-	ErrorNotAllowed            ErrorCondition = "amqp:not-allowed"
-	ErrorInvalidField          ErrorCondition = "amqp:invalid-field"
-	ErrorNotImplemented        ErrorCondition = "amqp:not-implemented"
-	ErrorResourceLocked        ErrorCondition = "amqp:resource-locked"
-	ErrorPreconditionFailed    ErrorCondition = "amqp:precondition-failed"
-	ErrorResourceDeleted       ErrorCondition = "amqp:resource-deleted"
-	ErrorIllegalState          ErrorCondition = "amqp:illegal-state"
-	ErrorFrameSizeTooSmall     ErrorCondition = "amqp:frame-size-too-small"
+	ErrCondInternalError         ErrCond = "amqp:internal-error"
```

Comment:
Nit: In a subsequent PR we should rearrange the lists so they are alphabetical.  I understand not doing it in this one if you don't want the diffs to be messy

Reply:
Good idea, sorted.